src/hub/Hub.sol

```diff
@@ -370,20 +370,13 @@ contract Hub is Multicall, Auth, Recoverable, IHub, IHubGatewayHandler {
     }
 
     /// @inheritdoc IHub
-    function addDebit(PoolId poolId, AccountId account, uint128 amount) external payable {
+    function updateJournal(PoolId poolId, JournalEntry[] memory debits, JournalEntry[] memory credits) external {
         _protected(poolId);
 
         accounting.unlock(poolId);
-        accounting.addDebit(account, amount);
-        accounting.lock();
-    }
 
-    /// @inheritdoc IHub
-    function addCredit(PoolId poolId, AccountId account, uint128 amount) external payable {
-        _protected(poolId);
+        _updateJournal(debits, credits);
```

Comment:
Maybe `_updateJournal` should just be a method exposed by `Accounting`? Would also make the `Hub` a bit smaller

Reply:
Interesting! If we can assume this will never will change. I agree in doing it

Reply:
@mustermeiszer thoughts?

Reply:
Ahhh. How do we envision calling into that from externals?